integration-tests/src/test/java/com/datastax/oss/driver/core/PoolBalancingIT.java

```diff
@@ -49,49 +49,56 @@
       SessionRule.builder(SIMULACRON_RULE)
           .withConfigLoader(
               DriverConfigLoader.programmaticBuilder()
-                  .withInt(DefaultDriverOption.CONNECTION_MAX_REQUESTS, CONNECTIONS)
-                  .withInt(DefaultDriverOption.CONNECTION_POOL_LOCAL_SIZE, REQUESTS_PER_CONNECTION)
+                  .withInt(DefaultDriverOption.CONNECTION_MAX_REQUESTS, REQUESTS_PER_CONNECTION)
+                  .withInt(DefaultDriverOption.CONNECTION_POOL_LOCAL_SIZE, POOL_SIZE)
                   .build())
           .build();
 
   @ClassRule
   public static final TestRule CHAIN = RuleChain.outerRule(SIMULACRON_RULE).around(SESSION_RULE);
 
+  private CountDownLatch done;
+  private AtomicReference<Throwable> unexpectedErrorRef;
+
+  @Before
+  public void setup() {
+    done = new CountDownLatch(1);
+    unexpectedErrorRef = new AtomicReference<>();
+  }
+
   @Test
   public void should_balance_requests_across_connections() throws InterruptedException {
-    CqlSession session = SESSION_RULE.session();
-
     // Generate just the right load to completely fill the pool. All requests should succeed.
-    int simultaneousRequests = CONNECTIONS * REQUESTS_PER_CONNECTION;
-    ExecutorService requestExecutor = Executors.newFixedThreadPool(simultaneousRequests);
-    AtomicReference<Throwable> unexpectedErrorRef = new AtomicReference<>();
-
-    ScheduledExecutorService timeoutExecutor = Executors.newScheduledThreadPool(1);
-    CountDownLatch done = new CountDownLatch(1);
+    int simultaneousRequests = POOL_SIZE * REQUESTS_PER_CONNECTION;
 
     for (int i = 0; i < simultaneousRequests; i++) {
-      requestExecutor.submit(
-          () -> {
-            while (!Thread.currentThread().isInterrupted()) {
-              try {
-                session.execute("SELECT release_version FROM system.local");
-              } catch (Throwable t) {
-                unexpectedErrorRef.compareAndSet(null, t);
-                // Even a single error is a failure, no need to continue
-                done.countDown();
-              }
-            }
-          });
+      reschedule(null, null);
     }
-    timeoutExecutor.schedule(done::countDown, 5, SECONDS);
-
-    done.await();
+    SECONDS.sleep(5);
```

Comment:
Introduction 5 seconds delay in a test seems like a lot, maybe instead we could create a new `AtomicInteger successfullRequests = simultaneousRequests`, increment it in every successful session execute, and change the sleep to:
`awaitUntil( () successfullRequests == simultaneousRequests)`? The test will keep the characteristic because we will be waiting until all succeed OR error will happen. wdyt?

Reply:
`simultaneousRequests` is the concurrent amount at any given point in time, we execute way more than that.

I'll lower the delay to 1 second, that's more than enough to reproduce the bug on 4.x.